R/Utilities.R

```diff
@@ -1012,38 +1052,74 @@ get_min_cluster_size <- function(tree, node) {
 
 #' Trivial distance function for arbitrary tree clustering
 #' 
-#' Trivial distance is defined as the difference in depths between the 
+#' Number of mutations along path from tip1 to LCA(tip1, tip2)
+#' Ensures if on same clade, join.
 #' 
 #' @param tree an object of class phylo
 #' @param tip1 the first leaf
 #' @param tip2 the second leaf
 #' @return the trivial distance between tip1, tip2
+#'
+#' @export
 trivial_dist <- function(tree, tip1, tip2) {
-    depths <- node.depth(tree)
-    edges <- tree$edge
-    path1 <- c(tip1)
-    root <- find_root(tree)
-    parent <- tip1
-    while (T) {
-      parent <- edges[, 1][edges[, 2] == parent]
-      path1 <- append(path1, parent)
-      if (parent == root) {
-          break
-      }
+  # node depths of tree
+  edges <- tree$edge
+  # Get the path from tip1 to root
+  path1 <- c(tip1)
+  root <- find_root(tree)
+  parent <- tip1
+  while (T) {
+    parent <- edges[, 1][edges[, 2] == parent]
+    path1 <- append(path1, parent)
+    if (parent == root) {
+      break
     }
-    
-    path2 <- c(tip2)
-    parent <- tip2
-    while (T) {
-      parent <- edges[, 1][edges[, 2] == parent]
-      path2 <- append(path2, parent)
-      if (parent == root || parent %in% path1) {
-        break
-      }
+  }
+  
+  mrca <- getMRCA(tree, c(tip1, tip2)) # MRCA of both
+  
+  
+  # Depths of the internal nodes that represent the parents of tip1, tip2 right before LCA
+  # ie the 'diverging' point or first split
+  path_length <- which(path1 == mrca)
+  
+  # Return the absolute difference between the depths
+  return(path_length)
+}
+
+
+
+
+
+#' Depth of tip1 parent immediately after LCA(tip1, tip2)
```

Comment:
Why do we want the depth of the parent immediately _after_ the LCA? Why can't we work with the LCA depth?

Reply:
Trying to use this so that merged clusters appear next to each other in the plotly graph.

If clades A, B and C all have LCA D, when we merge A and B for example we need to choose how to merge between A and B or A and C or B and C. The plotly tree is sorted by depth, so I was trying to use the depths of the node after LCA so one can distinguish between the three child clades. I am kind of souring on this idea though since it really is arbitrary, I still don't really know what the best solution for dealing with multifurcations is. Maybe we can have really small clusters (1/2/3 cells)?